app/services/rate_limiter.rb

```diff
@@ -69,24 +69,33 @@ def maxed?
     attempts && attempts >= RateLimiter.max_attempts(rate_limit_type)
   end
 
+  def expiration_minutes
+    minutes = self.class.attempt_window_in_minutes(rate_limit_type)
+    exponential_factor = self.class.attempt_window_exponential_factor(rate_limit_type)
+    attempt_window_max = self.class.attempt_window_max_in_minutes(rate_limit_type)
+    minutes *= exponential_factor ** (attempts - 1) if exponential_factor && attempts.positive?
+    if attempt_window_max && minutes > attempt_window_max
+      attempt_window_max
+    else
+      minutes
+    end
+  end
+
   def increment!
     return if limited?
-    value = nil
 
     now = Time.zone.now
+
+    @redis_attempts += 1
```

Comment:
I think we need to use the value returned from the Redis increment otherwise we risk a race-condition where two simultaneous each increment from 1 to 2 (even if Redis returns 3 for one of them).

Reply:
Hm, yeah, I'll think about this a bit more. This change was to address an off-by-one issue where `expiration_minutes` is hard to know to be based on current vs. in-progress-incremented attempts. Another way I'd thought about dealing with that was to have an optional `offset` parameter for the method, though maybe that could also be prone to race conditions.